transcom/mymove
# B-21488: UB Allowance Warning Int

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21488)

## Summary

In this PR implementation is being added to evaluate excess weight for a UB shipment. Excess weight can be evaluated on shipments at 2 key points. When a shipment is approved and when a shipment is updated. On approval if the prime estimated weight exist then it will be evaluated. On update if the updated shipment has a prime estimated weight included.

For UB shipments, based on the AC of the story, when the primeEstimatedWeight or the primeActualWeight is within 10% or greater of the UBAllowance then excessWeight will be triggered until it has been acknowledged. Same flow of all other shipments.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Ensure FEATURE_FLAG_UNACCOMPANIED_BAGGAGE is set to true
2. Determine a duty location you will use when creating a move. In the DB set that duty location address is_Oconus value to true
3.Create a move using that address discussed above as one of the duty locations
4. Create a ub shipment and add dependents
5. Complete the move creation process in milmove
6. As SC complete the counsel process for the move
7. As TOO approve the move. (Here excess weight will not be evaluated because primeEstimatedWeight is null at this time)
8. Look in the devs tools under entitlement object and take note of the ubAllowance. (Lets say it is 2000)
9. As Prime in the prime sim update the shipment adding a estimated weight that will trigger the excess weight(Lets say 1850). For ubAllowance of 2000 the threshold value is 1800. 2000 * .9 = 1800.
10. After updating the shipment view the move as TOO in the move queue. The status should say Approvals Requested.
![image](https://github.com/user-attachments/assets/8945e53c-9d4f-43d2-a8ed-9e8fbe0ff953)
11. Next as TOO view the move and under the Move Task Order tab and you should see the excess weight warning.
![image](https://github.com/user-attachments/assets/279cd61a-68e5-4429-94bc-defeba0c3b13)
12. Acknowledge the warning by clicking the "Review billable weight" link. You should notice the warning disappear
![image](https://github.com/user-attachments/assets/6d956ddd-a098-485a-a483-558f275b4656)
13. Next view the move in the move queue and you should see the status go back to approved
![image](https://github.com/user-attachments/assets/468168a9-e714-4d7f-a977-7e5e42f9eca4)


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
508 Compliance Check
![image](https://github.com/user-attachments/assets/719f76a2-dabb-4b33-989c-525c23dc8d46)
